CMakeLists.txt

```diff
@@ -21,6 +21,24 @@ file(READ version.txt PACKAGE_VERSION)
 string(STRIP ${PACKAGE_VERSION} PACKAGE_VERSION)
 message(STATUS "VowpalWabbit Version: ${PACKAGE_VERSION}")
 
+# Grab git commitish into variable
+IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
+  FIND_PACKAGE(Git)
+  IF(GIT_FOUND)
+    EXECUTE_PROCESS(
+      COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
+      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+      OUTPUT_VARIABLE "vw_GIT_COMMIT"
```

Comment:
Why lowercase `vw`?

Reply:
That seems to be how other libraries do it when they want to be able to have the files that define them be includable: <targetname>_<PROPERTY_NAME>.

See https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime.cmake#L30, though they also have it all uppercase. Having it be lower_UPPER_CAMEL does make it easier to segment it when reading quickly.

With all that said, I do not have a strong preference one way or the other, and am happy to change it.

Reply:
That's reasonable, happy to keep it how it is